### ISSUE TITLE
[DOC] net-ssh ed25519 support

### ISSUE THREAD

**Issue by mfazekas** [net-ssh/net-ssh#478]
While net-ssh 4.0 and later supports ed25519 out of the box, it does require `rbnacl` and other dependencies not required by net-ssh.

This issue documents the dependencies needed for ed25519 to work. TODO

**Comment by mattcg** [comment]
On Mac OS at least, you'll get this error if you don't `ssh-add [yourkey]` after system startup.

**Comment by mfazekas** [comment]
@mattcg can you ellaborate? Do you have bcrypt_pbkdf in your gemfile? What is the exact setup?

**Comment by mattcg** [comment]
I used `ssh-keygen` to generate a new ed25519 key with a passphrase. `net-ssh` will fail with the above error if I try, for example, a Capistrano deploy *unless* I use `ssh-add` to add the key to the agent (which prompts me for the password once).

After adding the key to the agent, I no longer see this error and SSH connections work perfectly. I *do not* have `bcrypt_pbkdf` in my Gemfile.

**Comment by mfazekas** [comment]
>On Mac OS at least, you'll get this error if you don't ssh-add [yourkey] after system startup.

@mattcg as noted in the description of this issue if you don't have `bcrypt_pbkdf` you will not be able to decrypt ed25519, and yes adding to agent will help.

**Comment by mwalters12** [comment]
I have bcrypt_pbkdf installed, but still seeing this error message. Any ideas?

Windows 10

```
ERROR: NotImplementedError: OpenSSH keys only supported if ED25519 is available
net-ssh requires the following gems for ed25519 support:
 * rbnacl (>= 3.2, < 5.0)
 * rbnacl-libsodium, if your system doesn't have libsodium installed.
 * bcrypt_pbkdf (>= 1.0, < 2.0)
See https://github.com/net-ssh/net-ssh/issues/478 for more information
LoadError : "Could not open library '.dll': The specified module could not be found.
"
```

**Comment by mfazekas** [comment]
@mwalters12 Can you open a separate issue, with details like full stacktrace? This issue tries to be a documentation issue, so i try to avoid too much conversation here.

**Comment by spuder** [comment]
To clarify, there are multiple ways to install the gems. 

Windows/linux

```
chef gem install rbnacl
chef gem install rbnacl-libsodium
chef gem install bcrypt_pbkdf
```

OSX users will proabably need sudo

```
sudo chef gem install rbnacl
sudo chef gem install rbnacl-libsodium
sudo chef gem install bcrypt_pbkdf
```

**Comment by Quintasan** [comment]
Any particular reason for `rbnacl < 5.0`? One would thought newer versions would work better.

**Comment by gregf** [comment]
I have run the following commands on my Linux box.

```
chef gem install rbnacl
chef gem install rbnacl-libsodium
chef gem install bcrypt_pbkdf
```

I'm still told these gems are missing even though I can run pry and load them manually.

**Comment by mattmartini** [comment]
What is the correct syntax for the Gemfile for multiple dependencies?

I'm getting a similar error message to @mwalters12:
`cap aborted!
NotImplementedError: unsupported key type `ssh-ed25519'
net-ssh requires the following gems for ed25519 support:
 * rbnacl (>= 3.2, < 5.0)
 * rbnacl-libsodium, if your system doesn't have libsodium installed.
 * bcrypt_pbkdf (>= 1.0, < 2.0)
See https://github.com/net-ssh/net-ssh/issues/478 for more information
Gem::LoadError : "can't activate rbnacl (< 5.0, >= 3.2.0), already activated rbnacl-5.0.0. Make sure all dependencies are added to Gemfile."
`

I put the following into my Gemfile:
`  gem 'rbnacl', '>= 3.2, < 5.0', :require => false
  gem 'rbnacl-libsodium', :require => false
  gem 'bcrypt_pbkdf', '>= 1.0, < 2.0', :require => false
`
The gems (and req false) were previously in the Gemfile.

When I added the version requirements, bundle install gives the following error:


`[!] There was an error parsing `Gemfile`: Illformed requirement [">= 3.2, < 5.0"]. Bundler cannot continue.

 #  from /Users/martini/Sites/RoR/GyroPalace/Gemfile:70
 #  -------------------------------------------
 #    gem 'capistrano-maintenance', '~> 1.0', :require => false
 >    gem 'rbnacl', '>= 3.2, < 5.0', :require => false
 #    gem 'rbnacl-libsodium', :require => false
 #  -------------------------------------------`

**Comment by aleksandrs-ledovskis** [comment]
@mattmartini See https://bundler.io/v1.15/man/gemfile.5.html#VERSION

**Comment by zzeroo** [comment]
These Gemfile entries works for me

```
gem 'rbnacl', '< 5.0', :require => false                                                                                                                                                                
gem 'rbnacl-libsodium', :require => false                                                                                                                                                               
gem 'bcrypt_pbkdf', '< 2.0', :require => false                                                                                                                                                          
```

**Comment by ayazahmadtarar** [comment]
> These Gemfile entries works for me
> 
> ```
> gem 'rbnacl', '< 5.0', :require => false                                                                                                                                                                
> gem 'rbnacl-libsodium', :require => false                                                                                                                                                               
> gem 'bcrypt_pbkdf', '< 2.0', :require => false                                                                                                                                                          
> ```

worked for me

**Comment by fathur-bookdoc** [comment]
> On Mac OS at least, you'll get this error if you don't `ssh-add [yourkey]` after system startup.

full command to save some googling (run this on your terminal):

`ssh-add -K ~/.ssh/id_rsa`

**Comment by hieunguyentrung** [comment]
Hi, @mattcg.
Thank you.
I can use this method `ssh-add `
But I was wondering, is there a way to solve it thoroughly, run only once? Because every time I want to deploy, I have to run this command again.

**Comment by HanyMoh** [comment]
> > On Mac OS at least, you'll get this error if you don't `ssh-add [yourkey]` after system startup.
> 
> full command to save some googling (run this on your terminal):
> 
> `ssh-add -K ~/.ssh/id_rsa`

Fixed :D
